core/cmd/shell_local.go

```diff
@@ -821,6 +824,45 @@ func dropDanglingTestDBs(lggr logger.Logger, db *sqlx.DB) (err error) {
 	return
 }
 
+type failedToRandomiseTestDBSequencesError struct{}
+
+func (m *failedToRandomiseTestDBSequencesError) Error() string {
+	return "failed to randomise test db sequences"
+}
+
+// randomiseTestDBSequences randomizes sequenced table columns sequence
+// This is necessary as to avoid false positives in some test cases.
+func randomiseTestDBSequences(db *sqlx.DB) error {
+	seqRows, err := db.Query(`SELECT sequence_schema, sequence_name FROM information_schema.sequences WHERE sequence_schema = $1`, "public")
+	if err != nil {
+		return fmt.Errorf("%s: error fetching sequences: %s", failedToRandomiseTestDBSequencesError{}, err)
+	}
+
+	defer seqRows.Close()
+	for seqRows.Next() {
+		var sequenceSchema, sequenceName string
+		if err = seqRows.Scan(&sequenceSchema, &sequenceName); err != nil {
+			return fmt.Errorf("%s: failed scanning sequence rows: %s", failedToRandomiseTestDBSequencesError{}, err)
+		}
+
+		var randNum *big.Int
+		randNum, err = crand.Int(crand.Reader, utils.NewBigI(10000).ToInt())
+		if err != nil {
+			return fmt.Errorf("%s: failed to generate random number", failedToRandomiseTestDBSequencesError{})
+		}
+
+		if _, err = db.Exec(fmt.Sprintf("ALTER SEQUENCE %s.%s RESTART WITH %d", sequenceSchema, sequenceName, randNum)); err != nil {
```

Comment:
Is there a way to do this natively, without `Sprintf`?

Reply:
Parametrised table names aren't supported :/. Although all of the values are pulled from the db itself (and rand) so this shouldn't be an issue

Reply:
I'm surprised the linter doesn't complain :shrug: